include/seqan/arg_parse/arg_parse_version_check.h

```diff
@@ -0,0 +1,466 @@
+{
+    if (!_checkWritability(me._path))
+    {
+#if defined(PLATFORM_WINDOWS)
+        TCHAR tmp_path [MAX_PATH];
+        if (GetTempPath (MAX_PATH, tmp_path) != 0)
+            me._path = tmp_path;
+        else //GetTempPath() returns 0 on failure
+            return false;
+# else // unix
+        me._path = "/tmp";
+#endif
+        me._updateCommand();
+    }
+
+    std::string version_file = me._path + "/" + me._name + "_version.txt";
+    double min_time_diff = 86400;                           // one day = 86400 seonds
+    double file_time_diff = _getFileTimeDiff(version_file); // time difference: last modified date until now in seconds
+    bool version_file_exists = (file_time_diff > -1) ? true : false;
+
+    if (file_time_diff < min_time_diff && version_file_exists)
+        return false; // only check for newer version once a day
+    
+    if (version_file_exists) // file exists. TODO:: ask if there should be a time limit here too
+    {
+        seqan::String<int> new_ver;
+        if (_readVersionNumbers(new_ver, version_file))
+        {
+            seqan::String<int> old_ver;
+            _getNumbersFromString(old_ver, me._version);
+
+            if (_isSmaller(old_ver, new_ver))
+            {
+                if(me._name == "seqan")
+                {
+                    std::cerr << "[SEQAN INFO] :: There is a newer SeqAn version available : SeqAn "
+                              << new_ver[0] << "." << new_ver[1] << "." << new_ver[2] << " Go to "
+                              << me._website << "\n"
+                              << "[SEQAN INFO] :: If you don't want to recieve this message again set --version-check APP_ONLY" << "\n\n";
+                }
+                else
+                {
+                    std::cerr << "[APP INFO] :: There is a newer version available: " << me._name << " " 
+                              << new_ver[0] << "." << new_ver[1] << "." << new_ver[2] << "\n"
+                              << "[APP INFO] :: Check out " << me._website << "\n"
+                              << "[APP INFO] :: If you don't want to recieve this message again set --version_check OFF" << "\n\n";
+                }
+            }
+            else if (_isSmaller(new_ver, old_ver))
+            {
+                std::cerr << "[APP INFO] :: We noticed your app version (" << me._version << ") is newer than the one registered.\n"
+                          << "[APP INFO] :: If you are the developer of this app, please send us an email to update your version info (support@seqan.de)" 
+                          << "[APP INFO] :: If not, you might want to contact the developer."
+                          << "\n\n";
+            }
+        }
+    }
+
+    if (me._program.empty())
+        return false;
+
+    // launch a seperate thread to not defer runtime
+    // std::cout << me._command << std::endl;
+    me._fut = std::async(std::launch::async, _callServer, me, version_file_exists);
```

Comment:
Instead of using this reference to a future thing, we could use a promise object.
The promise object creates a future which is stored in the ArgumentParser and is moved to the callable invoked by the thread. Then the promise sets the bool flag to inform the future. Like in the example of: http://en.cppreference.com/w/cpp/thread/promise